src/index.ts

```diff
@@ -73,187 +73,109 @@ export class PetraWallet implements AdapterPlugin {
-      }
-      return response as { hash: Types.HexEncodedBytes };
-    } catch (error: any) {
-      const errMsg = error.message;
-      throw errMsg;
-    }
+    const response = await this.provider!.signAndSubmitTransaction(
+      transaction,
+      options,
+    ).catch(remapPetraError);
+    return response as { hash: Types.HexEncodedBytes };
   }
 
   async signAndSubmitBCSTransaction(
     transaction: TxnBuilderTypes.TransactionPayload,
-    options?: any,
   ): Promise<{ hash: Types.HexEncodedBytes }> {
-    try {
-      const response = await this.provider?.signAndSubmitTransaction(
-        transaction,
-        options,
-      );
-      if ((response as AptosWalletErrorResult).code) {
-        throw new Error((response as AptosWalletErrorResult).message);
-      }
-      return response as { hash: Types.HexEncodedBytes };
-    } catch (error: any) {
-      const errMsg = error.message;
-      throw errMsg;
-    }
+    const response = await this.provider!.signAndSubmitTransaction(
+      transaction
+    ).catch(remapPetraError);
+    return response as { hash: Types.HexEncodedBytes };
   }
 
   async signMessage(message: SignMessagePayload): Promise<SignMessageResponse> {
-    try {
-      if (typeof message !== "object" || !message.nonce) {
-        throw `${PetraWalletName} Invalid signMessage Payload`;
-      }
-      const response = await this.provider?.signMessage(message);
-      if (response) {
-        return response;
-      } else {
-        throw `${PetraWalletName} Sign Message failed`;
-      }
-    } catch (error: any) {
-      const errMsg = error.message;
-      throw errMsg;
+    if (typeof message !== "object" || !message.nonce) {
+      throw `${PetraWalletName} Invalid signMessage Payload`;
     }
+    return this.provider!.signMessage(message).catch(remapPetraError);
   }
 
   async signTransaction(
     transactionOrPayload: Types.TransactionPayload | TxnBuilderTypes.TransactionPayload | AnyRawTransaction,
     optionsOrAsFeePayer?: TransactionOptions | boolean,
   ): Promise<AccountAuthenticator | Uint8Array> {
-    try {
-      // If "rawTransaction" is part of the args, then we have a v2 request
-      if ("rawTransaction" in transactionOrPayload) {
-        const transaction = transactionOrPayload;
-        const asFeePayer = (optionsOrAsFeePayer as boolean | undefined) ?? false;
-        const rawTxnV1 = convertV2toV1(transaction.rawTransaction, TxnBuilderTypes.RawTransaction);
+    // If "rawTransaction" is part of the args, then we have a v2 request
+    if ("rawTransaction" in transactionOrPayload) {
+      const transaction = transactionOrPayload;
+      const asFeePayer = (optionsOrAsFeePayer as boolean | undefined) ?? false;
+      const rawTxnV1 = convertV2toV1(transaction.rawTransaction, TxnBuilderTypes.RawTransaction);
+
+      const secondarySignersAddressesV1 = transaction.secondarySignerAddresses?.map(
+        (address) => convertV2toV1(address, TxnBuilderTypes.AccountAddress),
```

Comment:
confused. `convertV2toV1` implies that we convert v2 arguments to v1 arguments but the argument `TxnBuilderTypes` is a v1 type

Reply:
The function signature is

```typescript
convertV2toV1(src: SerializableV2, dst: DeserializableV1Class<SerializableV1>): SerializableV1;
```
we need to pass the class of the target object, since `deserialize` is a class method